File: sdks/java/endurance-tests/src/test/java/inc/reactor/sdk/endurance/Sample.java
package inc.reactor.sdk.endurance;

/**
 * One reading of the process, taken at the end of a cycle.
 *
 * @param cycle which cycle this was taken after
 * @param elapsedSeconds how long the run had been going
 * @param rssBytes resident set size — what the OS says this process is actually using
 * @param cpuNanos cumulative process CPU time; a raw counter, only meaningful as a delta
 * @param threads live threads
 * @param openFds open file descriptors, or -1 where the platform does not report them
 * @param liveClients SDK clients created and not yet closed
 * @param orphanedArenas arenas deliberately leaked because a callback was still running
 * @param nativeCommittedBytes committed native memory from NMT, or -1 when it is not enabled
 */
record Sample(
        int cycle,
        double elapsedSeconds,
        long rssBytes,
        long cpuNanos,
        int threads,
        long openFds,
        int liveClients,
        int orphanedArenas,
        long nativeCommittedBytes) {}

File: sdks/android/endurance-tests/src/main/kotlin/inc/reactor/sdk/android/endurance/Sample.kt
package inc.reactor.sdk.android.endurance

/**
 * One reading of the process, taken at the end of a cycle.
 *
 * @property cycle which cycle this was taken after
 * @property elapsedSeconds how long the run had been going
 * @property rssBytes resident set size — what the kernel says this process is actually using
 * @property cpuNanos cumulative process CPU time; a raw counter, only meaningful as a delta
 * @property threads live threads
 * @property openFds open file descriptors, or -1 where they cannot be counted
 * @property liveClients SDK clients created and not yet closed
 * @property orphanedGlobals JNI globals deliberately leaked because a callback was still running
 * @property nativeHeapBytes the native heap specifically, which on Android is the number that is
 *   not confounded by the managed heap moving underneath it. ART has no NMT; this is the
 *   equivalent question asked the way the platform can answer it.
 */
public data class Sample(
    val cycle: Int,
    val elapsedSeconds: Double,
    val rssBytes: Long,
    val cpuNanos: Long,
    val threads: Int,
    val openFds: Long,
    val liveClients: Int,
    val orphanedGlobals: Int,
    val nativeHeapBytes: Long,
)
